src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -169,6 +176,83 @@ public async Task<FormulaValue> EvalAsync(string expressionText, CancellationTok
             return await check.Expression.EvalAsync(parameters, cancel);
         }
 
+        /// <summary>
+        /// For private use because we don't want anyone defining a function without binding it.
+        /// </summary>
+        /// <returns></returns>
+        private UDFLazyBinder DefineFunction(UDFDefinition definition)
+        {
+            // $$$ Would be a good helper function 
+            var record = RecordType.Empty();
+            foreach (var p in definition.Parameters)
+            {
+                record = record.Add(p);
+            }
+
+            var check = new CheckWrapper(this, definition.Body, record);
+
+            var func = new UserDefinedTexlFunction(definition.Name, definition.ReturnType, definition.Parameters, check);
+            if (_customFuncs.ContainsKey(definition.Name))
+            {
+                return new UDFLazyBinder(

```

Comment:
> return new UDFLazyBinder( [](http://example.com/codeflow?start=15&length=26)

I recommend we throw since this can be considered a host error (not necessarily user error). 
The Host shouldn't have to scan the expressionError() list , those ought to be passed straight through to the user and should be things the user can directly fix (like syntax errors).

Reply:
Fixed